lib/ansible/plugins/lookup/cyberarkpassword.py

```diff
@@ -0,0 +1,140 @@
+
+            credential = ""
+            tmp_output, tmp_error = Popen(all_parms, stdout=PIPE, stderr=PIPE, stdin=PIPE).communicate()
+
+            if tmp_output:
+                credential = tmp_output
+
+            if tmp_error:
+                raise AnsibleError("ERROR => %s " % (tmp_error))
+
+            if credential and credential.endswith(b'\n'):
+                credential = credential[:-1]
+
+            output_names = self.output.split(",")
+            output_values = credential.split(self.delimiter)
+
+            for i in range(len(output_names)):
+                if output_names[i].startswith("passprops."):
+                    if "passprops" not in result_dict:
+                        result_dict["passprops"] = {}
+                    output_prop_name = output_names[i][10:]
+                    result_dict["passprops"][output_prop_name] = output_values[i]
+                else:
+                    result_dict[output_names[i]] = output_values[i]
+
+        except subprocess.CalledProcessError as e:
+            raise AnsibleError(e.output)
+        except OSError as e:
+            raise AnsibleError("ERROR - AIM not installed or clipasswordsdk not in standard location. ERROR=(%s) => %s " % (to_text(e.errno), e.strerror))
+
+        return [result_dict]
+
+
+class LookupModule(LookupBase):
+
+    """
+    USAGE: {{lookup('cyberarkpassword', appid='Application1', query='safe=Safe1;Folder=root;Object=User1',
+                    output='password,passprops.username,passprops.address' [, extra_parms])}}
+
+    It Requires CyberArk AIM Installed, and /opt/CARKaim/sdk/clipasswordsdk in place or set environment variable AIM_CLIPASSWORDSDK_CMD to the AIM
+    CLI Password SDK executable.
+
+     Args:
+         appid (str): Defines the unique ID of the application that is issuing the password request.
+         query (str): Describes the filter criteria for the password retrieval.
+         output (str): Specifies the desired output fields separated by commas. They could be: Password, PassProps.<property>, PasswordChangeInProcess
+         Optionally, you can specify extra parameters recognized by clipasswordsdk (like FailRequestOnPasswordChange, Queryformat, Reason, etc.)
+
+     Returns:
+         dict: A dictionary with 'password' as key for the credential, passprops.<property>, passwordchangeinprocess
+               If the specified property does not exist for this password, the value <na> will be returned for this property.
+               If the value of the specified property is empty, <null> will be returned.
+
+
+    for extra_parms values please check parameters for clipasswordsdk in CyberArk's "Credential Provider and ASCP Implementation Guide"
+
+    For Ansible on windows, please change the -parameters (-p, -d, and -o) to /parameters (/p, /d, and /o) and change the location of CLIPasswordSDK.exe
+
+    cyberarkpassword lookup can not be used as a source for loop
+    """
```

Comment:
We discussed the situation with the docs today as well.  We want this to get in before freeze and formatting of the docs in docs/docsite/rst/playbooks_lookups.rst will take some time and back and forth with @dharmabumstead to get right.  We also saw that a lot of the lookups are poorly documented on that page (they only have a single example in the section at the end instead of a section describing them in detail).  bcoca also said that he thinks he'll have the ability for documentation to be pulled out of plugins for ansble-doc and the website done for 2.4 release (so we'll target having in-plugin documentation for 2.5).

With all that in mind, go ahead and remove the documentation from playbooks_lookups.rst and we'll get that done right after 2.4 is released.